python_modules/dagster/dagster/_core/storage/event_log/in_memory.py

```diff
@@ -1,46 +1,57 @@
 import logging
-import time
-from collections import OrderedDict, defaultdict
-from typing import Dict, Iterable, Mapping, Optional, Sequence, Set, cast
+from collections import defaultdict
+from contextlib import contextmanager
+from typing import Callable
 
-import dagster._check as check
-from dagster._core.assets import AssetDetails
-from dagster._core.definitions.events import AssetKey
-from dagster._core.event_api import RunShardedEventsCursor
-from dagster._core.events import DagsterEventType
-from dagster._core.events.log import EventLogEntry
-from dagster._core.storage.event_log.base import AssetEntry, AssetRecord
+from sqlalchemy.pool import NullPool
+
+from dagster._core.storage.event_log.base import EventLogCursor
+from dagster._core.storage.sql import create_engine, get_alembic_config, stamp_alembic_rev
+from dagster._core.storage.sqlite import create_in_memory_conn_string
 from dagster._serdes import ConfigurableClass
-from dagster._utils import utc_datetime_from_timestamp
 
-from .base import (
-    EventLogConnection,
-    EventLogCursor,
-    EventLogCursorType,
-    EventLogRecord,
-    EventLogStorage,
-    EventRecordsFilter,
-)
+from .schema import SqlEventLogStorageMetadata
+from .sql_event_log import SqlEventLogStorage
 
 
-class InMemoryEventLogStorage(EventLogStorage, ConfigurableClass):
+class InMemoryEventLogStorage(SqlEventLogStorage, ConfigurableClass):
     """
     In memory only event log storage. Used by ephemeral DagsterInstance or for testing purposes.
 
     WARNING: Dagit and other core functionality will not work if this is used on a real DagsterInstance
     """
 
     def __init__(self, inst_data=None, preload=None):
-        self._logs = defaultdict(list)
-        self._handlers = defaultdict(set)
         self._inst_data = inst_data
-        self._wiped_asset_keys = defaultdict(float)
+        self._conn = self._create_connection()
+        self._handlers = defaultdict(set)
+        self._storage_id = 0  # mirror the storage id, to mimic watching cursors
+
+        self.reindex_events()
+        self.reindex_assets()
```

Comment:
do we need this?

Reply:
We have some checks to make sure that new DBs will have the secondary indexes marked correctly (for optimized queries)